<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The catch-all handler should come last in a chain of <code>catch</code> or <code>@catch</code> statements because it catches everything, and any more-specific <code>catch</code>/<code>@catch</code> that comes after it will never be used, even when the relevant condition occurs.</p>
</div>
<div class="paragraph">
<p>This C&#43;&#43; code sample is very similar to the Objective-C equivalent with <code>@try</code> and <code>@catch</code>.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">void f1()
{
  try
  {
    // ...
  }
  catch (...)
  {
    // Handle all exception types
  }
  catch (std::exception const &amp;e)  // Noncompliant - handler will never be called
  {
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">void f1()
{
  try
  {
    // ...
  }
  catch (std::exception const &amp;e)  // Compliant
  {
    // Handle standard exceptions
  }
  catch (...)        // Compliant catch-all handler
  {
    // Handle all other exception types
  }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C&#43;&#43;:2008, 15-3-7 - Where multiple handlers are provided in a single try-catch statement or function-try-block, any ellipsis (catch-all) handler shall occur last.</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s1045">is related to: <a data-rspec-id="S1045" class="rspec-auto-link">S1045</a></h3>

</div>
<div class="sect2">
<h3 id="_on_30_jan_2020_162541_nicolas_harraudeau_wrote">on 30 Jan 2020, 16:25:41 Nicolas Harraudeau wrote:</h3>
<div class="paragraph">
<p>This rule has little value for python as Python interpreter will raise a <code>SyntaxError</code>.</p>
</div>
</div>
</div>
</div>