<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>If the address of an automatic object is assigned to another automatic object of larger scope, or to a static object, or returned from a function then the object containing the address may exist beyond the time when the original object ceases to exist (and its address becomes invalid).</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">int* f(void) {
  int local_auto;
  return &amp;local_auto; // Noncompliant, returning address of an object allocated on the stack.
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>MISRA C:2004, 17.6</p>
</li>
<li>
<p>MISRA C&#43;&#43;:2008, 7-5-2</p>
</li>
<li>
<p>MISRA C:2012, 18.6</p>
</li>
<li>
<p><a href="https://wiki.sei.cmu.edu/confluence/x/UtcxBQ">CERT, DCL30-C.</a> - Declare objects with appropriate storage durations</p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>The address of 'xxx' is invalid once the function returns.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_duplicated_by_s838">is duplicated by: <a data-rspec-id="S838" class="rspec-auto-link">S838</a></h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s837">is related to: <a data-rspec-id="S837" class="rspec-auto-link">S837</a></h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s839">is related to: <a data-rspec-id="S839" class="rspec-auto-link">S839</a></h3>

</div>
</div>
</div>