<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>It should be clear to a casual reader what code a test is testing and what results are expected. Unfortunately, that&#8217;s not usually the case with the <code>ExpectedException</code> attribute since an exception could be thrown from almost any line in the method.</p>
</div>
<div class="paragraph">
<p>This rule detects MSTest and NUnit <code>ExpectedException</code> attribute.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vbnet" data-lang="vbnet">&lt;TestMethod&gt;
&lt;ExpectedException(GetType(ArgumentNullException))&gt; ' Noncompliant
Public Sub TestNullArg()
  '...
End Sub</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vbnet" data-lang="vbnet">&lt;TestMethod&gt;
Public Sub TestNullArg()
    Dim CallFailed As Boolean = False
    Try
    ' ...
    Catch ex As Exception
        CallFailed = true
    End Try
    Assert.IsTrue(CallFailed, "Expected call to MyMethod to fail with ArgumentNullException")
End Sub</code></pre>
</div>
</div>
<div class="paragraph">
<p>or</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vbnet" data-lang="vbnet">&lt;TestMethod&gt;
Public Sub TestNullArg()
    Assert.ThrowsException(Of ArgumentNullException)(Sub() ... )
End Sub</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>This rule ignores one-line test methods, since it is obvious in such methods where the exception is expected to be thrown.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s3658">relates to: <a data-rspec-id="S3658" class="rspec-auto-link">S3658</a></h3>

</div>
<div class="sect2">
<h3 id="_on_1_dec_2015_143700_tamas_vajk_wrote">on 1 Dec 2015, 14:37:00 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I changed minor things. It looks good.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_9_aug_2017_115944_amaury_levé_wrote">on 9 Aug 2017, 11:59:44 Amaury Levé wrote:</h3>
<div class="paragraph">
<p>Please review</p>
</div>
</div>
</div>
</div>