<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In Kotlin Gradle DSL a task can be defined in several ways:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>one way is to use <code>tasks.create(&#8230;&#8203;)</code> that eagerly configures it regardless if it&#8217;s required or not</p>
</li>
<li>
<p>another way is to use <code>tasks.register(&#8230;&#8203;)</code> that lazily configures the task only when it&#8217;s required: i.e. when it is located using query methods such as <code>TaskCollection.getByName(java.lang.String)</code>, or when it is added to the task graph for execution or when <code>Provider.get()</code> is called on the return value of this method.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>It is generally more efficient to use <code>tasks.register(&#8230;&#8203;)</code> instead of <code>tasks.create(&#8230;&#8203;)</code> as you won&#8217;t configure the task if you don&#8217;t need it.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Replace <code>tasks.create(&#8230;&#8203;)</code> with <code>tasks.register(&#8230;&#8203;)</code>.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">tasks.create("myTask") {
    group = JavaBasePlugin.DOCUMENTATION_GROUP
    description = "My task."
    // other configuration logic

    doLast {
      // ...
    }
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">tasks.register("myTask") {
    group = JavaBasePlugin.DOCUMENTATION_GROUP
    description = "My task."
    // other configuration logic

    doLast {
      // ...
    }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.gradle.org/current/javadoc/org/gradle/api/tasks/TaskContainer.html">org.gradle.api.tasks.TaskContainer</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_standards">Standards</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.gradle.org/current/userguide/task_configuration_avoidance.html">Task configuration avoidance</a></p>
</li>
<li>
<p><a href="https://www.youtube.com/watch?v=LvuVboGNVoE&amp;list=PL0UJI1nZ56yAHv9H9kZA6vat4N1kSRGis&amp;index=10">Configuration avoidance (Gradle best practice tip #8)</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>