<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Arrays in JavaScript have several methods for filtering, mapping, or folding that require a callback. Not having a return statement in such a callback function is most likely a mistake, because the processing of the array uses the return value of the callback. If there is no return, the callback will implicitly return <code>undefined</code>, which will likely fail.</p>
</div>
<div class="paragraph">
<p>This rule applies to the following methods of an array:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>Array.from</code></p>
</li>
<li>
<p><code>Array.prototype.every</code></p>
</li>
<li>
<p><code>Array.prototype.filter</code></p>
</li>
<li>
<p><code>Array.prototype.find</code></p>
</li>
<li>
<p><code>Array.prototype.findLast</code></p>
</li>
<li>
<p><code>Array.prototype.findIndex</code></p>
</li>
<li>
<p><code>Array.prototype.findLastIndex</code></p>
</li>
<li>
<p><code>Array.prototype.map</code></p>
</li>
<li>
<p><code>Array.prototype.flatMap</code></p>
</li>
<li>
<p><code>Array.prototype.reduce</code></p>
</li>
<li>
<p><code>Array.prototype.reduceRight</code></p>
</li>
<li>
<p><code>Array.prototype.some</code></p>
</li>
<li>
<p><code>Array.prototype.sort</code></p>
</li>
<li>
<p><code>Array.prototype.toSorted</code></p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">let arr = ["a", "b", "c"];
let merged = arr.reduce(function(a, b) {
  a.concat(b);
}); // Noncompliant: No return statement, will result in TypeError</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">let arr = ["a", "b", "c"];
let merged = arr.reduce(function(a, b) {
  return a.concat(b);
}); // merged === "abc"</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add a "return" statement to this callback.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_9_feb_2017_145423_carlo_bottiglieri_wrote">on 9 Feb 2017, 14:54:23 Carlo Bottiglieri wrote:</h3>
<div class="paragraph">
<p>Moved to Blocker severity, as the likehood is high, not low.</p>
</div>
</div>
</div>
</div>