<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Dictionary unpacking requires an object with methods <code>__getitem__</code> and <code>keys</code> or <code>__getitem__</code> and <code>__getattr__</code>. This is the case for any <a href="https://docs.python.org/3/glossary.html#term-mapping">mapping</a> object such as <code>dict</code>. Using an object which doesn&#8217;t have these methods will raise a <code>TypeError</code>.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">class A:
    pass

dict(**A())  # Noncompliant
{'a': 10, 'b': 20, **A()}  # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">class A:
  def __getitem__(self, key):
      return 2

  def keys(self):
      return ['1','2','3']

dict(**A())  # =&gt; {'1': 2, '2': 2, '3': 2}
{'a': 10, 'b': 20, **A()}  # =&gt; {'a': 10, 'b': 20, '1': 2, '2': 2, '3': 2}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>X is of type Y and cannot be unpacked with "**". Use a "mapping" object instead.</p>
</div>
</div>
</div>
</div>