<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Using a named constant to refer to an indicator makes the content of the field clearer, and therefore makes the code easier to read and maintain.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">      /Free

       If *In(25);  // Noncompliant
        // Process contents...
       EndIf;</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-rpg" data-lang="rpg">     D accountTotal    c                   25

      /Free

       If *In(accountTotal);
        // Process contents...
       EndIf;</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use a named constant for this reference to XXX.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_duplicated_by_s1977">is duplicated by: <a data-rspec-id="S1977" class="rspec-auto-link">S1977</a></h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s6164">relates to: <a data-rspec-id="S6164" class="rspec-auto-link">S6164</a></h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s2480">is related to: <a data-rspec-id="S2480" class="rspec-auto-link">S2480</a></h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s1583">is related to: <a data-rspec-id="S1583" class="rspec-auto-link">S1583</a></h3>

</div>
<div class="sect2">
<h3 id="_on_4_nov_2013_091654_freddy_mallet_wrote">on 4 Nov 2013, 09:16:54 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>Is implemented by https://jira.codehaus.org/browse/SONARJAVA-130</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_28_apr_2014_085008_dinesh_bolkensteyn_wrote">on 28 Apr 2014, 08:50:08 Dinesh Bolkensteyn wrote:</h3>
<div class="paragraph">
<p>In COBOL, this rule has been dropped in favor of "String literals should not be duplicated", see http://jira.sonarsource.com/browse/COBOL-997</p>
</div>
</div>
</div>
</div>