<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>WHERE</code> clause conditions that reinforce or contradict the definitions of their columns are useless; they are always either unconditionally true or unconditionally false. For instance, there&#8217;s no point in including <code>AND column IS NOT NULL</code> if the column is defined as non-null.</p>
</div>
<div class="paragraph">
<p><strong>Noteworthy</strong></p>
</div>
<div class="paragraph">
<p>This rule raises issues only when a <strong>Data Dictionary</strong> is provided during the analysis. SeeÂ https://docs.sonarqube.org/latest/analysis/languages/plsql/</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-sql" data-lang="sql">CREATE TABLE product
(id INT,
  name VARCHAR(6) NOT NULL,
  mfg_name VARCHAR(6),
  mfg_id INT
  ...

SELECT name, price
FROM product
WHERE name is not null -- Noncompliant; always true. This column is NOT NULL
  AND mfg_name = 'Too long name' -- Noncompliant; always false. This column can contain only 6 characters</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Based on the definition of "ccc", this condition is always ["TRUE"|"FALSE"].</p>
</li>
<li>
<p>Based on the constraints on "ccc", this condition is always ["TRUE"|"FALSE"].</p>
</li>
<li>
<p>Change this condition so that it does not always evaluate to ["TRUE"|"FALSE"].</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>problem condition</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s2583">relates to: <a data-rspec-id="S2583" class="rspec-auto-link">S2583</a></h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s3633">relates to: <a data-rspec-id="S3633" class="rspec-auto-link">S3633</a></h3>

</div>
<div class="sect2">
<h3 id="_on_17_jun_2016_150320_ann_campbell_wrote">on 17 Jun 2016, 15:03:20 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>from https://www.researchgate.net/publication/222432151_Semantic_errors_in_SQL_queries_A_quite_complete_list:</p>
</div>
<div class="paragraph">
<p>Error 27: Uncorrelated EXISTS-subqueries.If an EXISTS-subquery makes no reference to a tuple variable from theouter query, is is either globally true or globally false. This is a very unusualbehaviour. Actually, uncorrelated EXISTS-subqueries are simply missing joinconditions (possibly for anti-joins).</p>
</div>
<div class="paragraph">
<p>Not sure if we want to handle this here or in <a data-rspec-id="S3647" class="rspec-auto-link">RSPEC-3647</a></p>
</div>
</div>
</div>
</div>