<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In Kotlin, nullability is a part of the type system. This means that all types could be used in 2 ways. <code>T</code> - as non-nullable and <code>T?</code> as nullable. When you want to access properties or instance methods of a nullable type, you need to handle the potential null value. However, while accessing a non-nullable type, you don&#8217;t need to test nullability as the compiler knows that the value can never be <code>null</code>. So all the nullability checks on the non-nullable types are considered code smells.</p>
</div>
<div class="paragraph">
<p>Here is an example of non-nullable variable. <code>s</code> is of a type <code>String</code> and cannot be <code>null</code>.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">val s: String = ""</code></pre>
</div>
</div>
<div class="paragraph">
<p>Here is an example of nullable variable. Nullable variables are declared by using the <code>?</code>.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">val s: String? = null</code></pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Avoid using null checks on non-nullable variables.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="paragraph">
<p>If your variable type is non-nullable, any null checks are redundant. For example, <code>if (s == null) {}</code>, <code>requireNotNull(s)</code> and <code>checkNotNull(s)</code> can be dropped from your code.</p>
</div>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">val s: String = ""
if (s != null) { doSomething() } // This statement is always true</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">val s: String = ""
doSomething()</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">val s: String = ""
if (s == null) { doSomething() } // This statement is always false.</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">val s: String = ""</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_noncompliant_code_example_3">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">val s: String = "ABC"
return s ?: "" // This ?: is useless.</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_3">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">val s: String = "ABC"
return s</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_noncompliant_code_example_4">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">val s: String = ""
s!!.doSomething() //You do not need to assert that S is non-null.</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_4">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-kotlin" data-lang="kotlin">val s: String = ""
s.doSomething</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://kotlinlang.org/docs/null-safety.html#nullable-types-and-non-null-types">Kotlin Documentation - Null Safety</a></p>
</li>
<li>
<p><a href="https://kotlinlang.org/docs/strings.html">Kotlin Documentation - Strings</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_articles_blog_posts">Articles &amp; blog posts</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://blog.logrocket.com/complete-guide-null-safety-kotlin/">A complete guide to null safety in Kotlin</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>