<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>HTML &lt;table&gt; elements should not be used for layout purpose as it can confuse screen readers. It is recommended to use CSS instead.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue on every <code>&lt;table&gt;</code> element containing a <code>role</code>Â attribute set to <code>"presentation"</code> or <code>"none"</code>, which is how <a href="https://www.w3.org/WAI/tutorials/tables/tips/">W3C recommends</a> marks layout tables.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-html" data-lang="html">&lt;table role="presentation"&gt;
&lt;!-- ... --&gt;
&lt;/table&gt;</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://www.w3schools.com/html/html_layout.asp">HTML Layouts</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace this layout table with a CSS layout.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The opening &lt;table&gt; element, without its content.</p>
</div>
</div>
</div>
</div>