<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>One can use logical AND (<code>&amp;&amp;</code>) operator to conditionally render in React (aka short-circuit evaluation). <code>myCondition &amp;&amp; &lt;MyElement /&gt;</code> will return <code>&lt;MyElement /&gt;</code> if <code>myCondition</code> is <code>true</code> and <code>false</code> otherwise. As React renders boolean values to nothing, when condition is <code>false</code>, nothing will be rendered as intended. But if the condition has a non-boolean value which evaluates to <code>false</code> (e.g. <code>0</code>), that value will leak into the rendered result.</p>
</div>
<div class="paragraph">
<p>This rule will report when the condition has type <code>number</code> or <code>bigint</code>.</p>
</div>
<div class="paragraph">
<p>If React Native is used, the type <code>string</code> will also raise an error, as your render method will crash if you render <code>0</code>, <code>''</code>, or <code>NaN</code>.</p>
</div>
<div class="paragraph">
<p>Note that along with short-circuit evaluation, in order to achieve conditional rendering it&#8217;s possible to use ternary operator (<code>myCondition ? &lt;MyElement /&gt; : null</code>), which is less error-prone in this case as both return values are explicit.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">function Profile(props) {
  return &lt;div&gt;
    &lt;h1&gt;{ props.username }&lt;/h1&gt;
    { props.orders &amp;&amp; &lt;Orders /&gt; } { /* Noncompliant, 0 will be rendered if no orders available */ }
  &lt;/div&gt;
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">function Profile(props) {
  return &lt;div&gt;
    &lt;h1&gt;{ props.username }&lt;/h1&gt;
    { props.orders &gt; 0 &amp;&amp; &lt;Orders /&gt; }
  &lt;/div&gt;;
}</code></pre>
</div>
</div>
<div class="paragraph">
<p>or</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">function Profile(props) {
  return &lt;div&gt;
    &lt;h1&gt;{ props.username }&lt;/h1&gt;
    { props.orders ? &lt;Orders /&gt; : null }
  &lt;/div&gt;;
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://reactjs.org/docs/conditional-rendering.html">Conditional Rendering</a> - conditional rendering in React</p>
</li>
</ul>
</div>
</div>
</div>