<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>React components have a built-in <code>state</code> data. This data is used to store component property values. When <code>state</code> changes, the component is re-rendered. For functional components to manage <code>state</code>, React provides the <code>useState</code> hook, which returns the state value and a setter function to update its value.</p>
</div>
<div class="paragraph">
<p>When the setter function is called with the state variable as a parameter, nothing will happen. This can happen by mistake when attempting to reset a default value or invert a boolean, among others.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when calling the setter function with the state variable provided by the same <code>useState</code> React hook.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">import { useState } from "react";

function ShowLanguage() {
    const [language, setLanguage] = useState("fr-FR");
    return (
      &lt;section&gt;
        &lt;h1&gt;Your language is {language}!&lt;/h1&gt;
        &lt;button onClick={() =&gt; setLanguage(navigator.language)}&gt;Detect language&lt;/button&gt;
        &lt;button onClick={() =&gt; setLanguage(language)}&gt;Je préfère le Français&lt;/button&gt;{/* Non compliant: This button does nothing */}
      &lt;/section&gt;
    );
};</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">import { useState } from "react";

function ShowLanguage() {
    const [language, setLanguage] = useState("fr-FR");
    return (
      &lt;section&gt;
        &lt;h1&gt;Your language is {language}!&lt;/h1&gt;
        &lt;button onClick={() =&gt; setLanguage(navigator.language)}&gt;Detect language&lt;/button&gt;
        &lt;button onClick={() =&gt; setLanguage("fr-FR")}&gt;Je préfère le Français&lt;/button&gt;
      &lt;/section&gt;
    );
};</code></pre>
</div>
</div>
</div>
</div>
</div>