<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>If the denominator to a division or modulo operation is zero it would result in a fatal error.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>This is an issue because dividing by zero is a forbidden operation which leads to a fatal error.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>This issue can lead your program to an unexpected halt with all the inconveniences it entails.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Make sure that zero never reaches the denominator.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">def non_compliant():
    z = 0
    if (unknown()):
        # ...
        z = 4
    else:
        # ...
    z = 1 / z</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">def compliant():
    z = 0
    if (unknown()):
        # ...
        z = 4
    else:
        # ...
        z = 1
    z = 1 / z</code></pre>
</div>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_how_does_this_work">How does this work?</h3>
<div class="paragraph">
<p>By ensuring that for all the paths that can define the variable z, when none assigns it zero, we are sure that the issue is fixed.</p>
</div>
</div>
</div>
</div>