<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Generally it is not recommended to modify a collection while it is being iterated over. This may lead to exceptions and it could be the source of incorrect or unspecified behaviors in the code.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when a method modifies the size of a collection, while the same collection is iterated.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">def my_fun(my_dict):
    for key in my_dict:
        if key == 'foo':
            my_dict.pop(key) # Noncompliant</code></pre>
</div>
</div>
</div>
</div>
</div>