<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>A bare <code>raise</code> statement, i.e. a <code>raise</code> with no exception provided, will re-raise the last active exception in the current scope. If the "raise" statement is not in an <code>except</code> or <code>finally</code> block, no exception is active and a <code>RuntimeError</code> is raised instead.</p>
</div>
<div class="paragraph">
<p>If the bare <code>raise</code> statement is in a function called in an <code>except</code> statement, the exception caught by the <code>except</code> will be raised. This works but is hard to understand and maintain. Nothing indicates in the parent <code>except</code> that the exception will be reraised, and nothing prevents a developer from calling the function in another context.</p>
</div>
<div class="paragraph">
<p>Note also that using a bare <code>raise</code> in a <code>finally</code> block only works when an exception is active, i.e. when an exception from the <code>try</code> block is not caught or when an exception is raised by an <code>except</code> block. It will fail in any other case and should not be relied upon. This code smell is handled by rule <a data-rspec-id="S5704" class="rspec-auto-link">S5704</a>.</p>
</div>
<div class="paragraph">
<p>This rule raises an exception when a bare <code>raise</code> statement is not in an <code>except</code> or <code>finally</code> block.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">raise  # Noncompliant

def foo():
    raise  # Noncompliant
    try:
        raise  # Noncompliant
    except ValueError as e:
        handle_error()
    except:
        raise
    else:
        raise  # Noncompliant
    finally:
        raise

def handle_error():
    raise  # Noncompliant. This works but is hard to understand.</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">raise ValueError()

def foo():
    raise ValueError()
    try:
        raise ValueError()
    except:
        raise
    else:
        raise ValueError()
    finally:
        raise</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p>Python Documentation - <a href="https://docs.python.org/3/reference/simple_stmts.html#raise">The <code>raise</code> statement</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this "raise" statement or move it inside an "except" block.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The raise statement</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s5704">relates to: <a data-rspec-id="S5704" class="rspec-auto-link">S5704</a></h3>

</div>
<div class="sect2">
<h3 id="_relates_to_s5706">relates to: <a data-rspec-id="S5706" class="rspec-auto-link">S5706</a></h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s1039">is related to: <a data-rspec-id="S1039" class="rspec-auto-link">S1039</a></h3>

</div>
</div>
</div>