<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Classes subclassing only <code>unittest.TestCase</code> are considered test cases, otherwise they might be mixins.</p>
</div>
<div class="paragraph">
<p>As classes subclassing <code>unittest.TestCase</code> will be executed as tests, they should define test methods and not be used as "abstract" parent helper. Methods within the class will be discovered by the test runner if their name starts with <code>test</code>. If a method intended to be a test does not respect this convention, it will not be executed.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when a method is not discoverable as a test and is never used within its test case class.</p>
</div>
<div class="paragraph">
<p>This rule will not raise if:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>The method is called directly from another method.</p>
</li>
<li>
<p>The method overrides an existing one in <code>unittest.TestCase</code>  (example: a <code>tearDown</code>  method).</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import unittest
class MyTest(unittest.TestCase):
  def setUp(self): ... # OK (unittest.TestCase method)
  def something_test(self): ... # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">import unittest
class MyTest(unittest.TestCase):
  def setUp(self): ...
  def test_something(self): ...</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Rename this method so that it starts with "test" or remove this unused helper.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Primary: the name of the method.</p>
</div>
</div>
</div>
</div>