<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Setting the environment variable using the <code>ENV</code> instruction creates a new layer in the Docker image.
Calling <code>RUN unset &lt;env-variable&gt;</code> creates a new layer, but the environment variable can be still dumped from the previous layer.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The environment variables often contain secrets, tokens, and other sensitive information.
They are present in the containers and could be dumped anytime.
Calling <code>unset</code> doesn&#8217;t prevent this information from being hidden for other commands.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>To prevent runtime access, the environment variable can be set and unset in a single <code>RUN</code> instruction.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-docker" data-lang="docker">ENV $ADMIN_USER
RUN unset $ADMIN_USER</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-docker" data-lang="docker">RUN export ADMIN_USER="admin" \
    &amp;&amp; ... \
    &amp;&amp; unset ADMIN_USER</code></pre>
</div>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_how_does_this_work">How does this work?</h3>
<div class="paragraph">
<p>In this example, the visibility of <code>ADMIN_USER</code> is only limited to the single layer.
However, the value can still be extracted from the image.
The best solution is to use <code>ARG</code> instead.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.docker.com/engine/reference/builder/#env">ENV - Dockerfile reference</a></p>
</li>
<li>
<p><a href="https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#env">ENV - Best practices for writing Dockerfiles</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Use the ARG instruction or set &amp; unset environment variable in a single layer.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Highlight the <code>unset</code> command with the environment variable.</p>
</div>
<hr>
</div>
</div>
</div>