<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Unpacking lets developers assign values of an iterable simultaneously to different targets in a single assignment.
However, for this assignment to work, the iterable should have the same number of elements as there are targets in the target list.
If this is not respected, a <code>ValueError</code> will be produced at runtime.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">def foo(param):
  ls = [1, 2, 3]
  x, y = ls # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">def foo(param):
  ls = [1, 2, 3]
  x, y, z = ls</code></pre>
</div>
</div>
</div>
</div>
</div>