<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Variables can be declared with or without types. Variables declared without a type will be implicitly typed if the declaration includes an initialization, and the compiler then type-checks every usage of typed variables. On the other hand, a variable declared with the <code>any</code> type explicitly instructs the compiler not to do any type-checking, which is risky.</p>
</div>
<div class="paragraph">
<p>The <code>unknown</code> type should be preferred over the <code>any</code> type if itâ€™s not possible to have a better typing. <code>any</code> fully disables type-checking allowing to do anything with the variable, while <code>unknown</code> requires to narrow the type before doing something specific.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">let a = 42;  // implicitly typed to number
let b: number = 42;  // explicitly typed to number
let c: any = 42;  // Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">let a = 42;
let b: number = 42;
let c: number = 42;</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="ulist">
<ul>
<li>
<p>Remove this use of "any" so that implicit typing will be used.</p>
</li>
<li>
<p>Replace "any" with a specific type.</p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p><code>any</code></p>
</div>
</div>
</div>
</div>