<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>There is no good reason to create a new object to not do anything with it. Most of the time, this is due to a missing piece of code and so could lead to an unexpected behavior in production.</p>
</div>
<div class="paragraph">
<p>If it was done on purpose because the constructor has side-effects, then that side-effect code should be moved into a separate method and called directly.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">new MyConstructor(); // Non-Compliant</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">var something = new MyConstructor();  // Compliant</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>Immediately dropped new objects inside <code>try</code>-statements are ignored.</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">try {
  new MyConstructor();
} catch (e) {
  /* ... */
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Either remove this useless object instantiation of "XXXX" or use it</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_27_may_2015_115701_ann_campbell_wrote">on 27 May 2015, 11:57:01 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Looks good [~freddy.mallet]. I forgot that moving this to a sub-task wouldn&#8217;t clear out the extra fields. Thanks for taking care of that.</p>
</div>
</div>
<div class="sect2">
<h3 id="_is_related_to_s905">is related to: <a data-rspec-id="S905" class="rspec-auto-link">S905</a></h3>

</div>
<div class="sect2">
<h3 id="_on_21_aug_2015_152029_freddy_mallet_wrote">on 21 Aug 2015, 15:20:29 Freddy Mallet wrote:</h3>
<div class="paragraph">
<p>This RSPEC should be part of <a data-rspec-id="S905" class="rspec-auto-link">RSPEC-905</a> but as this seems to be impossible to not generate some false-positives, we prefer to keep it isolated.</p>
</div>
</div>
</div>
</div>