<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Trying to access a dictionary key that does not exist will raise a <code>KeyError</code>.</p>
</div>
<div class="paragraph">
<p>When trying to access or remove a key that may not be there, several solutions are possible:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Use the <code>get()</code> method instead of a subscription. It will return <code>None</code> for a missing key instead of raising a <code>KeyError</code>.</p>
</li>
<li>
<p>Use the <code>setdefault()</code> method to provide keys that may be missing with a default value.</p>
</li>
<li>
<p>Check that the key is present in the dictionary with the <code>if key in dict:</code> construct.</p>
</li>
<li>
<p>Use a <code>try/except</code> block and handle the potential <code>KeyError</code>.</p>
</li>
<li>
<p>Use a <code>defaultdict</code> instead of a regular <code>dict</code> object.</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">def foo():
    my_dict = {'k1': 42}
    ...
    value = my_dict['k2']  # Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-python" data-lang="python">def foo():
    my_dict = {'k1': 42}
    ...
    if 'k2' in my_dict:
        value = my_dict['k2']</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Fix this access on a dictionary key that may be missing.
Fix this "pop" operation on a dictionary key that may be missing.
Fix this "del" operation on a dictionary key that may be missing.</p>
</div>
<hr>
</div>
</div>
</div>