<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>There are situations where <code>super()</code> must be invoked and situations where <code>super()</code> cannot be invoked.</p>
</div>
<div class="paragraph">
<p>The basic rule is: a constructor in a non-derived class cannot invoke <code>super()</code>; a constructor in a derived class must invoke <code>super()</code>.</p>
</div>
<div class="paragraph">
<p>Furthermore:</p>
</div>
<div class="ulist">
<ul>
<li>
<p><code>super()</code> must be invoked before the <code>this</code> and <code>super</code> keywords can be used.</p>
</li>
<li>
<p><code>super()</code> must be invoked with the same number of arguments as the base class' constructor.</p>
</li>
<li>
<p><code>super()</code> can only be invoked in a constructor - not in any other method.</p>
</li>
<li>
<p><code>super()</code> cannot be invoked multiple times in the same constructor.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Some issues are not raised if the base class is not defined in the same file as the current class. This is a known limitation for this rule.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">class Dog extends Animal {
  constructor(name) {
    super();
    this.name = name;
    super();         // Noncompliant
    super.doSomething();
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">class Dog extends Animal {
  constructor(name) {
    super();
    this.name = name;
    super.doSomething();
  }
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Lacked a call of 'super()' in some code paths.</p>
</div>
<div class="paragraph">
<p>Expected to call 'super()'.</p>
</div>
<div class="paragraph">
<p>Unexpected duplicate 'super()'.</p>
</div>
<div class="paragraph">
<p>Unexpected 'super()' because 'super' is not a constructor.</p>
</div>
<div class="paragraph">
<p>Unexpected 'super()'.</p>
</div>
<div class="paragraph">
<p>'super'|'this' is not allowed before 'super()'.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_supercedes_s3833">supercedes: <a data-rspec-id="S3833" class="rspec-auto-link">S3833</a></h3>

</div>
</div>
</div>