<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p><code>setuid</code> family of functions are used to change the user identity of the caller process in order to change privileges for the following actions. If the function call returns with an error, the rest of the program will execute with unexpected privileges leading to unexpected behaviour.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">void f() {
  ...
  setuid(0); // Noncompliant
  ...
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">void f() {
  ...
  if (setuid(0)) {
    // fail
  }
  ...
}</code></pre>
</div>
</div>
</div>
</div>
</div>