<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The Dockerfile should contain at most one <code>ENTRYPOINT</code> and one <code>CMD</code> instruction, because only the last one will have an effect.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Multiple <code>ENTRYPOINT</code> or <code>CMD</code> instructions in a file can lead to confusion as we may think they are all applied, which is not the case.
Previous <code>ENTRYPOINT</code> and <code>CMD</code> instructions should be removed to avoid this.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-docker" data-lang="docker">ENTRYPOINT ignored_entrypoint param1 param2
ENTRYPOINT effective_entrypoint param1 param2

CMD ignored_cmd param1 param2
CMD effective_cmd param1 param2</code></pre>
</div>
</div>
<div class="paragraph">
<p>Here we have multiple <code>ENTRYPOINT</code> and <code>CMD</code> instructions.
The first <code>ENTRYPOINT</code> and the first <code>CMD</code> instructions will have no effect.
Although this is valid in Docker, this can lead to confusion and be error-prone, as we may expect each <code>CMD</code> and <code>ENTRYPOINT</code> to have an effect.</p>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-docker" data-lang="docker">ENTRYPOINT effective_entrypoint param1 param2

CMD effective_cmd param1 param2</code></pre>
</div>
</div>
<div class="paragraph">
<p>Here we have only one ENTRYPOINT and one CMD instruction.
Each of them will be considered by the docker container and have a normal effect as we can expect.</p>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.docker.com/engine/reference/builder/#entrypoint">ENTRYPOINT - Dockerfile reference</a></p>
</li>
<li>
<p><a href="https://docs.docker.com/engine/reference/builder/#cmd">CMD - Dockerfile reference</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this <code>CMD</code> or <code>ENTRYPOINT</code> instruction which will be ignored.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>Highlight the entire <code>CMD</code> and <code>ENTRYPOINT</code> instruction.</p>
</div>
<hr>
</div>
</div>
</div>